Sources/Turf/Geometries/LineString.swift

```diff
@@ -308,4 +308,25 @@ extension LineString {
         // Ported from https://github.com/Turfjs/turf/blob/4e8342acb1dbd099f5e91c8ee27f05fb2647ee1b/packages/turf-simplify/lib/simplify.js
         coordinates = Simplifier.simplify(coordinates, tolerance: tolerance, highestQuality: highestQuality)
     }
+    
+    /**
+     Returns all intersections with another `LineString`.
+     
+     This function is roughly equivalent to the [turf-line-intersect](https://turfjs.org/docs/#lineIntersect) package of Turf.js ([source code](https://github.com/Turfjs/turf/tree/master/packages/turf-line-intersect/)).
+     */
+    public func intersection(with line: LineString) -> [LocationCoordinate2D]? {
+        var intersections: [String : LocationCoordinate2D] = [:]
+        zip(coordinates.dropLast(), coordinates.dropFirst()).forEach { segment1 in
+            zip(line.coordinates.dropLast(), line.coordinates.dropFirst()).forEach { segment2 in
+                if let intersection = Turf.intersection(LineSegment(segment1.0, segment1.1),
+                                                        LineSegment(segment2.0, segment2.1)) {
+                    let key = "\(intersection.latitude),\(intersection.longitude)"
```

Comment:
A string representation is kind of weird, but it’s probably the most straightforward approach until apple/swift#37398 lands, since we wouldn’t be able to have `LocationCoordinate2D` privately conform to `Hashable`.

Reply:
I'm a bit concerned about floating-point accuracy here. And about time complexity constant.
What if we implement a small struct like
```swift
private struct HashableCoordinate: Hashable {
    var latitude: Double
    var longitude: Double
}
```
And use `Set<HashableCoordinate>` that is then converted to an array of coordinates.
Or is it too much? May it be useful somewhere else in Turf?

Reply:
Yes, that would work. Pedantically, Turf should represent GeoJSON Point objects as its own hashable `Point` type instead of reusing `CLLocationCoordinate2D`. We stuck to the latter type as a practical matter, to avoid excessive conversions to and from a very common type. This intersection method’s performance would be dominated by the non-RTree-based intersection search, but you have a point that the struct representation would give us more certainty.

Reply:
Implemented a Hashable wrapper